### PR TITLE
Adding a table for currently connected WiFi information

### PR DESCRIPTION
**Pull request by jedi22**
A simple table using CoreWLAN to get the status of all wifi interfaces. Also refactored out extractSsid because both wifi_networks and wifi_status need it.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 10f9695066c25451e631caf151a92a28849aa1c7 (Job results: [759](https://jenkins.osquery.io/job/osqueryPullRequestAudit/759/)) failed the code audit or documentation test.

**Comment by jedi22** [comment]
I believe this table will break compilation on 10.9 as it uses CWWiFiClient which is listed at 10.10+

**Comment by osqueryer** [comment]
:no_good: The commit 6c8bc2cfd47cd65344b6c631f16de65ba4b635fa (Job results: [760](https://jenkins.osquery.io/job/osqueryPullRequestAudit/760/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 2f85d1afa1d6b520929d82e0a6e286ecc133dd40 (Job results: [761](https://jenkins.osquery.io/job/osqueryPullRequestAudit/761/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit c326b3a85bce8093ba977af1121c6af7c28a8d71 (Job results: [762](https://jenkins.osquery.io/job/osqueryPullRequestAudit/762/)) failed the code audit or documentation test.

**Comment by theopolis** [comment]
Nice! Does this pass a:
```
./tools/analysis/profile.py --leaks --shell ./build/darwin10.12/osquery/osqueryi --query "select * from wifi_status"
```

**Comment by jedi22** [comment]
./tools/analysis/profile.py --leaks --shell ./build/darwin10.12/osquery/osqueryi --query "select * from wifi_status"
Analyzing leaks in query: select * from wifi_status
  definitely: 0 leaks for 0 total leaked bytes.

File: osquery/tables/networking/darwin/wifi_status.mm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  QueryData results;
+  std::string interfaceName = "en0";
+  NSArray<CWInterface*>* interfaces =
+      [[CWWiFiClient sharedWiFiClient] interfaces];
+  if (interfaces == nil || [interfaces count] == 0) {
+    return results;
+  }
+  for (CWInterface* interface in interfaces) {
+    Row r;
+    NSString* strptr;
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_status.mm]: Why let this dangle uninitialized?

File: osquery/tables/networking/darwin/wifi_status.mm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      r["bssid"] = std::string([strptr UTF8String]);
+    }
+    strptr = [interface ssid];
+    if (strptr != nil) {
+      r["network_name"] = std::string([strptr UTF8String]);
+    }
+    r["rssi"] = INTEGER([interface rssiValue]);
+    r["noise"] = INTEGER([interface noiseMeasurement]);
+    r["security_type"] = getSecurityName([interface security]);
+    CWChannel* cwc = [interface wlanChannel];
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_status.mm]: Could this be nil?

File: osquery/tables/networking/darwin/wifi_utils.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <CoreWLAN/CoreWLAN.h>
+#include <iomanip>
+// SSIDs have no character set associated with them
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_utils.h]: These doc blocks are incorrect. Take a look at the other headers for reference (in ./include/osquery).

File: osquery/tables/networking/darwin/wifi_utils.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <CoreWLAN/CoreWLAN.h>
+#include <iomanip>
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_utils.h]: Nit, space between OS includes and C++ includes, and space between the includes and start of code.

File: osquery/tables/networking/darwin/wifi_utils.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+// SSIDs have no character set associated with them
+// mirror Apple's representation of them
+std::string extractSsid(const CFDataRef& data);
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_utils.h]: Did you intend to put these symbols within the osquery::tables namespace?

File: osquery/tables/networking/darwin/wifi_utils.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+// Change an Apple Constant into a human readable string representing
+// the network encryption type
+std::string getSecurityName(CWSecurity cw);
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_utils.h]: What type is this input argument? Can you improve performance by passing a const reference?

File: osquery/tables/networking/darwin/wifi_utils.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+// Change an Apple constant into the channel width
+int getChannelWidth(CWChannel* cwc);
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_utils.h]: Can these raw pointers be const?

File: osquery/tables/networking/darwin/wifi_utils.mm [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <CoreFoundation/CoreFoundation.h>
+
+#include <osquery/system.h>
+
+#include "osquery/tables/networking/darwin/wifi_utils.h"
+
+std::string extractSsid(const CFDataRef& data) {
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi_utils.mm]: Missing namespace?

File: osquery/tables/networking/darwin/wifi_utils.h (hunk 6 above)
Comment (jedi22) [inline osquery/tables/networking/darwin/wifi_utils.h]: CWSecurity is an enum

**Comment by osqueryer** [comment]
:no_good: The commit b0052b487ec93422a1c443af088fb906cc1a412c (Job results: [765](https://jenkins.osquery.io/job/osqueryPullRequestAudit/765/)) failed the code audit or documentation test.

**Pull request merged**